### DISCUSSION TITLE
How can i use multiple persistentMarkers?

### DISCUSSION THREAD

**Discussion by chaeeunDev** [patrykandpatrick/vico#921] [Questions] [answered]
### Question

i want to use 3 or more persistentMarkers.
how can i use it?

persistentMarkers = rememberExtraLambda( ) { marker at index_x}

### Vico version(s)

2.0.0-alpha.28

**Comment by peshkira** [comment, marked as answer]
Hey!

You should be able to add multiple markers in the lambda.

```
persistentMarkers = rememberExtraLambda( ) { 
    marker1 at x1
    marker2 at x2
    ...
}
```

where `marker1` and `marker2` are instances of your markers and `x1` and `x2` are x coordinates.
I believe the marker appears only if there is an actual `y` value at the given `x` coordinate. If there is none, you don't see a marker afaik.

More info can be found in the release notes here: https://patrykandpatrick.com/vico/releases/2.0.0-alpha.23
Not sure if the documentation has been updated yet to reflect on this.

**Reply by chaeeunDev** [reply]
Oh! It works!!! Thank you for your kind reply @peshkira

**Reply by Gowsky** [reply]
> If there is none, you don't see a marker afaik.

That's right.

> Not sure if the documentation has been updated yet to reflect on this.

Yes, the documentation is up to date. `at` is listed [here](https://patrykandpatrick.com/vico/api/2.0.0-beta.2/core/com.patrykandpatrick.vico.core.cartesian/-cartesian-chart/-persistent-marker-scope/). It's not explicitly stated that you can call it multiple times because that's a basic feature of functions in general. However, we plan on adding a usage example that demonstrates this.

**Reply by marcosalis** [reply]
> Yes, the documentation is up to date. at is listed [here](https://patrykandpatrick.com/vico/api/2.0.0-beta.2/core/com.patrykandpatrick.vico.core.cartesian/-cartesian-chart/-persistent-marker-scope/).

Unfortunately that link is broken.
Could you please expand on the `rememberExtraLambda` delegate? It doesn't seem to exist in the latest library's source code.

**Reply by Gowsky** [reply]
Yes, it doesn't exist anymore. Please see [here](https://github.com/patrykandpatrick/vico/discussions/1033#discussioncomment-12594928).

**Comment by marcosalis** [comment]
> I believe the marker appears only if there is an actual y value at the given x coordinate. If there is none, you don't see a marker afaik.

I'm running into this "issue" (perhaps expected behaviour?). I have a:
```
 lineSeries { series(x, y) }
```

Where `y` contains a single element list with `0` (I'm using a line cartesian chart as a single dimension graph, where all values on the `y` axis are zero). I'm trying to add persistent markers at different positions for a few `x` values, but it won't add them for `x` values that don't have a corresponding `y`. Why is that? I thought that the behaviour of `series` would simply reuse the same `y` value if the provided list size is less than the one for the `x` axis.

**Reply by Gowsky** [reply]
>  I thought that the behaviour of `series` would simply reuse the same `y` value if the provided list size is less than the one for the `x` axis.

By definition, the data of a Cartesian chart is a list of _xy_ pairs, so the _x_ and _y_ series must be of the same size (as stated in the documentation). The displayed data exactly matches what you enter. An _x_ value without a _y_ value is meaningless, and there's no reason for the library to assume that it should be linked to some particular _y_ value.

> Where `y` contains a single element list with `0`

If so, no line is displayed. At least two points must be connected to form a line.

> Why is that?

Markers are displayed only for values that exist in the dataset.
